localflavor/md/validators.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import re
+
+from django.core.exceptions import ValidationError
+from django.core.validators import RegexValidator
+from django.utils.translation import ugettext_lazy as _
+
+
+class MDIDNOFieldValidator(RegexValidator):
+    """
+    Validation for Moldavian IDNO.
+
+    .. versionadded:: 1.6
+    """
+
+    error_message = _('Enter a valid IDNO number.')
+
+    def __init__(self):
+        super(MDIDNOFieldValidator, self).__init__(regex='^\d{13}$',
+                                                   message=self.error_message)
```

Comment:
It's better to just override the `regex` and `message` variables from the `RegexValidator` instead of setting them manually in `__init__`.

I read a translation of this site that says an IDNO is an alpha numeric value. Is the regex correct (only digits)?
https://servicii.gov.md/ServiceDetails.aspx?id=5b54755b-174c-4576-a8a4-5df336eca90d

Reply:
For sure IDNO is only numeric, its a company id, not sure about IDNP (personal).
In our case its only companies ids

Reply:
Ok, the numberic regex is fine. I trust you know how it's used.

I still think you should override the `regex` and `message` variables from the `RegexValidator` instead of setting them manually in `__init__`. For example:

```
class MDIDNOFieldValidator(RegexValidator):
     regex = '^\d{13}$'
     message = _('Enter a valid IDNO number.')
```